### PR TITLE
DevUI: Allow regex for defined allowed hosts

### PR DESCRIPTION
**Pull request by phillip-kruger**
Fix comment in https://github.com/quarkusio/quarkus/pull/40979#issuecomment-2293669897

@slallemand please can you test if this solve your issue. 

Here is an example usage. Let's say you want to match all subdomains in foo.bar (so http://*.foo.bar) you will use `^([a-zA-Z0-9-]+\.)*foo\.bar(/.*)?$`

Let me know

### CODE REVIEW

**Comment by slallemand** [comment]
Not sure how to test that :-/

**Comment by phillip-kruger** [comment]
Will using a regex as described above work for you ?

**Comment by slallemand** [comment]
Yes, that looks good to me 👍

**Review by cescoffier — APPROVED** [review]

File: extensions/vertx-http/deployment/src/main/java/io/quarkus/devui/deployment/DevUIConfig.java [hunk 1]
```diff
@@ -26,6 +26,7 @@ public class DevUIConfig {
      * More hosts allowed for Dev UI
      *
      * Comma separated list of valid URLs, e.g.: www.quarkus.io, myhost.com
+     * (This can also be a regex)
```
Comment (gsmet) [inline extensions/vertx-http/deployment/src/main/java/io/quarkus/devui/deployment/DevUIConfig.java:29]: I think this is a bit brittle. And given it's a security feature, I'm a bit worried about it.

The fact that anything being a valid pattern makes it a pattern means that `my.host.com` will be a Pattern. Meaning that `my2host.com` will match.

I think we need to be a lot more careful about it and take a step back.

I don't know how it's handled in other areas where we have this kind of thing.

/cc @cescoffier

Comment (gsmet) [reply extensions/vertx-http/deployment/src/main/java/io/quarkus/devui/deployment/DevUIConfig.java:29]: I dropped the `triage/backport` label for now. We need to discuss this.

Comment (cescoffier) [reply extensions/vertx-http/deployment/src/main/java/io/quarkus/devui/deployment/DevUIConfig.java:29]: Well could reduce this and use just a prefix (with a startsWith and not a regexp)

Comment (gsmet) [reply extensions/vertx-http/deployment/src/main/java/io/quarkus/devui/deployment/DevUIConfig.java:29]: I don't know what we're doing for the CORS for instance? We should probably try to be consistent.

Comment (phillip-kruger) [reply extensions/vertx-http/deployment/src/main/java/io/quarkus/devui/deployment/DevUIConfig.java:29]: @gsmet I first look for a complete match (like we always have done) before I then check for a regex. Regex has to a valid regex (starting with ^)

**Pull request merged**